### PR TITLE
Add network feature

### PR DESCRIPTION
**Pull request by wei-lee**
add new function to allow users to upload notes to the backend server

### CODE REVIEW

**Comment by wei-lee** [comment]
ping @TommyJ1994

File: app/src/main/res/layout/fragment_network.xml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    <Button
+        android:id="@+id/cancelUploadBtn"
+        style="@style/Widget.AppCompat.Button.Colored"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_alignParentBottom="true"
+        android:layout_alignParentStart="true"
+        android:layout_marginBottom="@dimen/authentication_button_margin_bottom"
+        android:enabled="false"
```
Comment (tomjackman) [inline app/src/main/res/layout/fragment_network.xml]: I think this should be `true` by default? The cancel button is only being made visible during the upload process, but the button is not being enabled.

File: app/src/main/java/com/feedhenry/securenativeandroidtemplate/features/network/presenters/UploadNotesPresenter.java [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+        ClientStatus(boolean hasPerm, long numberOfNotes) {
+            this.hasPermission = hasPerm;
+            this.numberOfNotes = numberOfNotes;
+        }
+    }
+
+    private class ClientStatusCheckerTask extends AsyncTask<Void, Void, ClientStatus> {
+
+        private Exception error;
+        String requiredRole = Constants.ACCESS_CONTROL_ROLES.ROLE_MOBILE_USER;
```
Comment (tomjackman) [inline app/src/main/java/com/feedhenry/securenativeandroidtemplate/features/network/presenters/UploadNotesPresenter.java]: Maybe we can use the `api-access` role for interactions with the API? - [keycloak/secure-app-realm.json](https://github.com/feedhenry/mobile-security/blob/master/projects/keycloak/secure-app-realm.json#L189-L196). The secure-user, secure-user2 and superuser are in the API Access Keycloak group so those users can interact with the API and upload notes. The API Server note route would need to be protected by this role then too to limit API access to only users in that group with something like: `keycloak.protect('realm:api-access')`

**Review by tomjackman — COMMENTED** [review]
Looks Good :+1:

File: app/src/main/res/layout/fragment_network.xml (hunk 1 above)
Comment (wei-lee) [reply app/src/main/res/layout/fragment_network.xml]: yes, it should be `true`

File: app/src/main/java/com/feedhenry/securenativeandroidtemplate/features/network/presenters/UploadNotesPresenter.java (hunk 2 above)
Comment (wei-lee) [reply app/src/main/java/com/feedhenry/securenativeandroidtemplate/features/network/presenters/UploadNotesPresenter.java]: yep, good point. will change this

**Review by tomjackman — APPROVED** [review]

**Pull request merged**